exercises/src/main/scala/fpinscala/errorhandling/Either.scala

```diff
@@ -1,24 +1,55 @@
 package fpinscala.errorhandling
 
-
 import scala.{Option => _, Either => _, Left => _, Right => _, _} // hide std library `Option` and `Either`, since we are writing our own in this chapter
 
 sealed trait Either[+E,+A] {
- def map[B](f: A => B): Either[E, B] = sys.error("todo")
+  def map[B](f: A => B): Either[E, B] =
+    this match {
+      case Right(a) => Right(f(a))
+      case Left(e) => Left(e)
+    }
+
+  def flatMap[EE >: E, B](f: A => Either[EE, B]): Either[EE, B] =
+    map(f) match {
```

Comment:
Very nice. Funny thing: this is exactly what Martin Odersky talked about in his lecture on monads in the Reactive Programming course.